File: Stage14/Java/Problem1620.java
package Stage14.Java;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.HashMap;
import java.util.StringTokenizer;

public class Problem1620 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        StringTokenizer st = new StringTokenizer(br.readLine());

        int N = Integer.parseInt(st.nextToken());
        int M = Integer.parseInt(st.nextToken());

        HashMap<String, Integer> nameToNum = new HashMap<>();
        String[] numToName = new String[N + 1];

        for (int i = 1; i <= N; i++) {
            String name = br.readLine();
            nameToNum.put(name, i);
            numToName[i] = name;
        }

        for (int i = 0; i < M; i++) {
            String q = br.readLine();
            if (Character.isDigit(q.charAt(0))) {
                int num = Integer.parseInt(q);
                sb.append(numToName[num]).append('\n');
            } else {
                sb.append(nameToNum.get(q)).append('\n');
            }
        }

        System.out.print(sb);
        br.close();
    }
}

File: Stage14/Python/Problem1620.py
import sys
input = sys.stdin.readline

def main():
    N, M = map(int, input().split())
    nameToNum = {}
    numToName = {}
    
    for i in range(1, N + 1):
        name = input().strip()
        nameToNum[name] = i
        numToName[str(i)] = name
    
    for _ in range(M):
        q = input().strip()
        if q[0].isdigit():
            print(numToName[q])
        else:
            print(nameToNum[q])
    
if __name__ == "__main__":
    main()